Client/multiplayer_sa/CMultiplayerSA_WidescreenFix.cpp

```diff
@@ -0,0 +1,1111 @@
+    fHUDHeight[93] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[94] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[95] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[96] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[97] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[98] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[99] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[100] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[101] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[102] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[103] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[104] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[105] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+
+    fHUDHeight[14] /= 1.125f;
+    fHUDHeight[15] /= 1.125f;
+    fHUDHeight[16] /= 1.125f;
+}
+
+void UpdateScreenAspectRatio()
+{
+    fWideScreenWidthScale = 640.0f / (*(pfScreenAspectRatio) * 448.0f);
+    fWideScreenHeightScale = 448.0 / 448.0f;
+    fWideScreenWidthProperScale = static_cast<float>(pRsGlobal->width) / (*pfScreenAspectRatio * 448.0f);
+    fWideScreenHeightProperScale = static_cast<float>(pRsGlobal->height) / 448.0f;
+    fDefaultWidth = (4.0f / 3.0f);
+
+    UpdateFrontendFixes();
+    UpdateMiscFixes();
+    UpdateHUDFixes();
+    UpdateScriptFixes();
+}
+
+void InstallAspectRatioFixes()
+{
+    MemPut<BYTE>(0x745BD1, 0x7D);
+    MemPut<BYTE>(0x745BD9, 0x7C);
+
+    // Proportional coronas
+    MemSet((void*)0x6FB2C9, 0x90, 4);
+
+    MemPut<BYTE>(0x6FB2BD, 0x6C);
+    MemPut<BYTE>(0x6FB2DC, 0x78);
+    MemPut<BYTE>(0x713BE5, 0x20);
+    MemPut<BYTE>(0x713B6D, 0x38);
+    MemPut<BYTE>(0x713CFB, 0x38);
+    MemPut<BYTE>(0x713EFC, 0x30);
+    MemPut<BYTE>(0x714004, 0x38);
+
+    HookInstall(0x6FF420, (DWORD)CalculateAspectRatio, 5);
+
+    HookInstallCall(0x0053D695 - 0x1, (DWORD)UpdateScreenAspectRatio);
+    HookInstallCall(0x0053D7B2 - 0x1, (DWORD)UpdateScreenAspectRatio);
+    HookInstallCall(0x0053D967 - 0x1, (DWORD)UpdateScreenAspectRatio);
+    HookInstallCall(0x0053E771 - 0x1, (DWORD)UpdateScreenAspectRatio);
+    HookInstallCall(0x0053EB1A - 0x1, (DWORD)UpdateScreenAspectRatio);
+}
+
+void GetMemoryAddresses()
+{
+    pRsGlobal = (RsGlobal*)0xC17040;
+    pfScreenAspectRatio = (float*)0xC3EFA4;
+}
+
+float __stdcall StretchXHook(float fValue)
+{
+    if (pRsGlobal->width == 640)
+        return fValue;
+    else
+        return (float)pRsGlobal->width * fValue * 0.0015625f;
+}
+
+float __stdcall StretchYHook(float fValue)
+{
+    if (pRsGlobal->height == 448)
+        return fValue;
+    else
+        return (float)pRsGlobal->height * fValue * 0.002232143f;
+}
```

Comment:
You can use the conditional operator (? :) here for better readability. What do you think about static_cast here?

Reply:
it will be not used anymore, static cast would be better tho